### PR TITLE
Fix some valgrind warnings 

### PR DESCRIPTION
**Pull request by pavelmash**
This patch allow to run mormot based app under varglind - `$define HASAESNI` should be commented for this.

Also fixed potential issue in UnCamelCase (as reported by valgrind)

### CODE REVIEW

File: SynCommons.pas [hunk 1]
```diff
@@ -37500,7 +37500,7 @@ function UnCamelCase(D, P: PUTF8Char): integer;
   Space := D;
   DBeg := D;
   SpaceBeg := D;
-  if (D<>nil) and (P<>nil) then // avoid GPF
+  if (D<>nil) and (P<>nil) then begin // avoid GPF
   repeat
```
Comment (synopse) [inline SynCommons.pas]: please add properly the indentation spaces needed by the new begin..end block. ;)

Comment (pavelmash) [reply SynCommons.pas]: I do not add indentation to simplify a review for a maintainer - if I will diff become non-readible :)

Comment (pavelmash) [reply SynCommons.pas]: I'm add an indentation.  Actually I'm use gitlab in my every day work and miss a "hide whitespace changes" feature in github diff view

Comment (synopse) [reply SynCommons.pas]: indentation allows direct merge - I don't have problem to do the proper review even if github doesn't handle it well, but I would like to be able to make a direct merge :)

**Pull request merged**